tokio/src/io/util/vec_with_initialized.rs

```diff
@@ -107,8 +107,8 @@ where
         // Note that this relies on the fact that `V` is either `Vec<u8>` or
         // `&mut Vec<u8>`, so the vector returned by `self.vec.as_mut()` cannot
         // change from call to call.
+        self.num_initialized = parts.initialized;
         unsafe {
-            self.num_initialized = parts.initialized;
             vec.set_len(parts.len);
         }
```

Comment:
I'm not so sure that this makes the code easier to read. The assignment to `num_initialized` is important for the safety of the `set_len` call too.

Reply:
I partially disagree, as checking for soundness includes more than just looking at the unsafe blocks. If there's an `unsafe` around the entire function code, it's easy to deduct that everything in the function has to be considered for soundness.